Source/Scene/FrameState.js

```diff
@@ -150,6 +150,19 @@ define([
         this.creditDisplay = creditDisplay;
 
         /**
+         * An array of functions to be called at the start of the frame.  This array
+         * will be cleared at the start of each frame.
+         *
+         * @type {FrameState~BeforeRenderCallback[]}
+         *
+         * @example
+         * frameState.beforeRender.push(function() {
+         *   // take some action, raise an event, etc.
+         * });
+         */
+        this.beforeRender = [];
```

Comment:
Did you talk to @lilleyse about this?  We have similar concepts elsewhere and I'm not sure that we should add another.

The code is also tightly coupled meaning that, for example, when looking at the timer query object, it looked like it would never be removed from `beforeRender` because that logic is in `Scene`.  Sometimes this is unavoidable, but it should be thought through carefully so we do not tangle the rendering engine into something that we can't maintain.